pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/channel/MailboxContentStreamObserver.java

```diff
@@ -75,7 +98,9 @@ public MailboxContentStreamObserver(GrpcMailboxService mailboxService,
       StreamObserver<Mailbox.MailboxStatus> responseObserver, boolean isEnabledFeedback) {
     _mailboxService = mailboxService;
     _responseObserver = responseObserver;
-    _receivingBuffer = new ArrayBlockingQueue<>(DEFAULT_MAILBOX_QUEUE_CAPACITY);
+    // TODO: Replace unbounded queue with bounded queue when we have backpressure in place.
+    // It is possible this will create high memory pressure since we have memory leak issues.
+    _receivingBuffer = new LinkedBlockingQueue();
```

Comment:
I took a deeper look today since it has been a while for me. I think for us this will worsen the memory issues we have seen substantially.

A common edge case that can be hit is: the sender took a while to reach MailboxSendOperator, and by that time MailboxReceive on the receiver has already hung up. In this case the sender will still end up initiating the connection and sending the data, filling up this queue but the queue will never be polled. Since there's a leak in the MailboxService maps, this will lead to large leaks.

Let's discuss this today if possible.

Reply:
This doesn't make it worse because it is always this way. We were simply dropping data block and returning the wrong result. The memory situation is always there. Also the in memory mailbox is using the same work around today. We have to have back pressure in place to fix this issue.

Reply:
Making the queue unbounded means that if the sender sends 20 blocks then all of them would be leaked to memory. At present after the 5th block that's sent, the subsequent sent blocks won't be added to the queue and hopefully not leaked either.

The blocks are leaked because the map in Mailbox service keeps a reference to the GrpcReceivingMailbox that keeps a reference to the MailboxContentStreamObserver, which keeps a reference to the `_receivingBuffer` queue.

This is based on my understanding.. lmk if there's something wrong

Reply:
> We were simply dropping data block and returning the wrong result.

I don't think a wrong result would be returned. Instead the errorContent would be set using the error lock and subsequent polls on `MailboxContentStreamObserver` would return the error.

Can you confirm how a wrong result can be returned?

Reply:
I added returning error code to make sure we are not returning wrong result... 

but we are not able to finish the query and returning error most of the time as long as data load is large. we have observed this issue a lot in our testing environment. 

Also as said in the PR description, this is a temp fix. We need backpressure and memory leak fix in place, which I am working on, to fix the underlying problem.